src/lib.rs

```diff
@@ -1109,6 +1109,125 @@ console.log(a);
     );
   }
 
+  #[tokio::test]
+  async fn test_build_graph_jsx_import_source_types() {
```

Comment:
Looks good. In case you want to add more tests for this, doing it and expanding on the functionality in `tests/specs/graph` is probably easier than manually setting up a loader and such (there is a concept of "spec options" where adding compiler options could probably be done).